release/selinux.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package release
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/selinux"
+)
+
+// SELinuxLevelType encodes the state of SELinux support found on this system.
+type SELinuxLevelType int
+
+const (
+	// NoSELinux indicates that SELinux is not enabled
+	NoSELinux SELinuxLevelType = iota
+	// SELinux is supported and in permissive mode
+	SELinuxPermissive
+	// SELinux is supported and in enforcing mode
+	SELinuxEnforcing
+)
+
+var (
+	selinuxLevel   SELinuxLevelType
+	selinuxSummary string
+
+	selinuxIsEnabled   = selinux.IsEnabled
+	selinuxIsEnforcing = selinux.IsEnforcing
+)
+
+func init() {
+	selinuxLevel, selinuxSummary = probeSELinux()
```

Comment:
Can this be done lazily please

Reply:
Why?

Reply:
Because importing release doesn't mean you need to know the status of SELinux or Apparmor or Seccomp or anything else. There are functions for that.

Reply:
What makes me think all the slow, feature testing pieces, should probably be in a separate package.

Reply:
I tried that with apparmor making it a separate package but we ended up putting it here in release.go.